Veir/Benchmarks.lean

```diff
@@ -443,21 +443,20 @@ builtin.module() {
 #eval! testBench "add-zero-forwards" 10
 
 /--
-info:
-builtin.module() {
+info: builtin.module() {
   ^2:
     %3 = arith.constant 42 ⏎
-    %34 = arith.addi(%3, %3)
-    %33 = arith.addi(%34, %34)
-    %32 = arith.addi(%33, %33)
-    %31 = arith.addi(%32, %32)
-    %30 = arith.addi(%31, %31)
-    %29 = arith.addi(%30, %30)
-    %28 = arith.addi(%29, %29)
+    %25 = arith.addi(%3, %3)
+    %28 = arith.addi(%25, %25)
     %27 = arith.addi(%28, %28)
-    %26 = arith.addi(%27, %27)
-    %25 = arith.addi(%26, %26)
-    test.test(%25)
+    %31 = arith.addi(%27, %27)
+    %29 = arith.addi(%31, %31)
+    %30 = arith.addi(%29, %29)
+    %33 = arith.addi(%30, %30)
+    %32 = arith.addi(%33, %33)
+    %34 = arith.addi(%32, %32)
+    %26 = arith.addi(%34, %34)
+    test.test(%26)
```

Comment:
Is this related? Otherwise, maybe a separate PR?

Reply:
That's fine. A 1-sentence explanation in the PR message will avoid confusion when people later on look at this commit.

For example: 

> We update a VEIR benchmark, as the changed key type in our hashmaps results in a change in rewrite order on our IR. We will resolve this in a future commit when we finalize the rewriter.

Reply:
This is fixed by using the same Hash as `Nat`